File: uri/ad-hock/rabiola/1876.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    string vine;
    
    while (cin >> vine) {
        vector<int> segments;
        int count = 0;
        
        for (char c : vine) {
            if (c == 'o') {
                count++;
            } else {
                if (count > 0) {
                    segments.push_back(count);
                    count = 0;
                }
            }
        }
        
        if (count > 0) {
            segments.push_back(count);
        }
        
        int maxSize = 0;
        
        for (int i = 0; i < segments.size(); i++) {
            int size;
            if (i == 0 || i == segments.size() - 1) {
                size = segments[i];
            } else {
                size = segments[i] / 2;
            }
            maxSize = max(maxSize, size);
        }
        
        cout << maxSize << endl;
    }
    
    return 0;
}

File: uri/ad-hock/rabiola/1876.rs
use std::io::{self, BufRead};

fn main() {
    let stdin = io::stdin();
    
    for line in stdin.lock().lines() {
        let vine = line.unwrap();
        let mut segments = Vec::new();
        let mut count = 0;
        
        for c in vine.chars() {
            if c == 'o' {
                count += 1;
            } else {
                if count > 0 {
                    segments.push(count);
                    count = 0;
                }
            }
        }
        
        if count > 0 {
            segments.push(count);
        }
        
        let mut max_size = 0;
        
        for (i, &segment) in segments.iter().enumerate() {
            let size = if i == 0 || i == segments.len() - 1 {
                segment
            } else {
                segment / 2
            };
            max_size = max_size.max(size);
        }
        
        println!("{}", max_size);
    }
}

